tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Service/Microsoft.Legal.MatterCenter.Repository/Matter/MatterRepository.cs

```diff
@@ -1082,21 +1083,29 @@ public bool SetItemPermission(ClientContext clientContext, IList<IList<string>>
                                 {
                                     // Removing permission for all the user except current user with full control
                                     // Add those users in list, then traverse the list and removing all users from that list
-                                    
+
                                     //Get email from the user instead of name for comparison
                                     Principal principal = role.Member;
-                                    User currentListUser = (User)principal;
-                                    if (currentListUser.Email.ToLower() == user.ToLower() && !((role.Member.Title.ToLower() == loggedInUserTitle.ToLower()) && (roleDef.Name ==
-                                        matterSettings.EditMatterAllowedPermissionLevel)))
+                                    if (principal.GetType().Name != "Group")
                                     {
-                                        roleDefinationList.Add(roleDef);
+                                        User currentListUser = (User)principal;
+                                        if (currentListUser.Email.ToLower() == user.ToLower() && !((role.Member.Title.ToLower() == loggedInUserTitle.ToLower()) && (roleDef.Name ==
+                                            matterSettings.EditMatterAllowedPermissionLevel)))
+                                        {
+                                            roleDefinationList.Add(roleDef);
+                                        }
                                     }
+
                                 }
-                                foreach (RoleDefinition roleDef in roleDefinationList)
+                                if (roleDefinationList.Count > 0)
```

Comment:
@premchandpl What is the purpose of this condition? The foreach will loop if there are values and otherwise it exit.